lib/services/linter-warning-service.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Emits warnings for Linter-related events.
+ * Similar to the WarningService, but kept in a separate module that only includes methods used by the Linter
+ * to keep the browser bundle size small.
+ * @author Francesco Trotta
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * A service that emits warnings for Linter-related events.
+ */
+class LinterWarningService {
+	/**
+	 * A function called internally to emit warnings using the Node.js API, if available.
+	 * @type {((warning: string, type: string) => void) | undefined}
+	 * @protected
+	 */
+	_emitWarning = globalThis.process?.emitWarning;
```

Comment:
It seems like this is done so that you can extend the class and still call `emitWarning`? If so, I'd say just make `emitWarning` a public method instead. I'd like to get rid of all these leading underscores. :smile: